### PR TITLE
Adds hamburger-button

### PR DESCRIPTION
**Pull request by ehannes**
Adds a hamburger button that you can use together with for instance `oma-menu`.

### CODE REVIEW

File: components/hamburger-button/src/index.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  }, [])
+
+  return html`
+    <style>
+      .hamburger-button {
+        background: none;
+        border: none;
+        cursor: pointer;
+        display: block;
+        height: 48px; /* Minimum size for buttons according to Lighthouse */
```
Comment (d-Pixie) [inline components/hamburger-button/src/index.js:34]: Should be `min-height` in that case.

File: components/hamburger-button/src/index.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        justify-self: var(--menu__justify-button, unset);
+        margin: 0;
+        padding: 0;
+        width: 48px; /* Minimum size for buttons according to Lighthouse */
```
Comment (d-Pixie) [inline components/hamburger-button/src/index.js:38]: `min-width`

File: components/hamburger-button/src/index.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        z-index: 5;
+      }
+
+      .hamburger-button__lines {
+        display: block;
+        display: flex;
+        flex-direction: column;
+        height: 26px;
+        justify-content: space-between;
+        margin-left: auto; /* Adjust lines to page margin */
```
Comment (d-Pixie) [inline components/hamburger-button/src/index.js:48]: That comment means nothing to me...

**Review by d-Pixie — CHANGES_REQUESTED** [review]
I think this should be part of them menu, or actually be the meny. The button will never be used in any other circumstance - given that it has a hardcoded appearance - and the full menu functionality is not far from this anyway.

**Comment by ehannes** [comment]
As you said before, let's make this the default button component in `oma-menu`. I'll wait adapting this PR until we have merged the menu component.

File: components/hamburger-button/src/index.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+const HamburgerButton = (element) => {
+  /*
+   * Query the button element from the shadow DOM.
+   *
+   * Since the button is added in the component,
+   * this query must be run when the component has been rendered.
+   **/
+  const queryButton = () =>
+    element.shadowRoot.querySelector('.hamburger-button')
```
Comment (ehannes) [inline components/hamburger-button/src/index.js:11]: consider using `querySelector('slot')`

File: components/hamburger-button/src/index.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+      .hamburger-button__line--middle {
+        transition: transform 0.2s ease-in-out;
+      }
+
+      .hamburger-button__line--bottom {
+        transform-origin: 0% 100%;
+        transition: transform 0.4s ease-in-out;
+      }
+
+      .hamburger-button--active
```
Comment (ehannes) [inline components/hamburger-button/src/index.js:75]:
```
slot[data-open='true'] {
...
}
```

File: components/hamburger-button/src/index.js [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+   * this query must be run when the component has been rendered.
+   **/
+  const queryButton = () =>
+    element.shadowRoot.querySelector('.hamburger-button')
+
+  useEffect(() => {
+    const button = queryButton()
+
+    const onClick = () => {
+      button.classList.toggle('hamburger-button--active')
```
Comment (ehannes) [inline components/hamburger-button/src/index.js:17]: Set `data-open = true`

File: components/menu/src/index.js [hunk 7]
```diff
@@ -1,9 +1,110 @@
...
+        ></span>
+      </div>
+    </button>
+  `
+}
+
+customElements.define('oma-hamburger-button', component(HamburgerButton))
+
 const Menu = (element) => {
+  const slotButton = element.querySelector('[slot=button]') !== null
```
Comment (d-Pixie) [inline components/menu/src/index.js:94]: Rename to `has_user_provided_button` or something.

File: components/menu/src/index.js [hunk 8]
```diff
@@ -1,9 +1,110 @@
...
+        <slot name="button"></slot>
+      `
+    : html`
+        <oma-hamburger-button></oma-hamburger-button>
+      `
+
   useEffect(() => {
     const shadowRoot = element.shadowRoot
-    const button = shadowRoot.querySelector('slot[name=button]')
+    const button = slotButton
```
Comment (ehannes) [inline components/menu/src/index.js:105]: This should of course be renamed!

File: components/menu/src/index.js [hunk 9]
```diff
@@ -1,9 +1,110 @@
...
+      </div>
+    </button>
+  `
+}
+
+customElements.define('oma-hamburger-button', component(HamburgerButton))
+
 const Menu = (element) => {
+  const slotButton = element.querySelector('[slot=button]') !== null
+  const button = slotButton
```
Comment (d-Pixie) [inline components/menu/src/index.js:95]: Rename to `rendered_button`

Comment (ehannes) [inline components/menu/src/index.js:95]: `renderedButton`

File: components/menu/src/index.js [hunk 10]
```diff
@@ -1,9 +1,110 @@
...
+    ? html`
+        <slot name="button"></slot>
+      `
+    : html`
+        <oma-hamburger-button></oma-hamburger-button>
```
Comment (d-Pixie) [inline components/menu/src/index.js:100]:
```
const onClick = () => {
  panel.toggleAttribute('data-open')
}
const button = slotButton ? html`
        <slot name="button" @click="${onClick}></slot>
      `
    : html`
        <oma-hamburger-button @click="${onClick}></oma-hamburger-button>
     `
```

Comment (d-Pixie) [reply components/menu/src/index.js:100]: Maybe ... https://lit.dev/docs/v1/components/events/

File: components/menu/src/index.js (hunk 8 above)
Comment (d-Pixie) [inline components/menu/src/index.js:105]: Should be able to delete this.

**Review by d-Pixie — CHANGES_REQUESTED** [review]

File: components/menu/src/index.js (hunk 8 above)
Comment (d-Pixie) [inline components/menu/src/index.js:105]: Generally try to keep the conditional part as small as possible.
```
const button = shadowRoot.querySelector( slotButton ? 'slot[name=button]' : 'oma-hamburger-button')
```

**Review by d-Pixie — APPROVED** [review]

**Pull request merged**